src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1274,14 +1274,20 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 
 		// Respect option to reveal an editor if it is already visible
 		if (options && options.revealIfVisible) {
-			const editorsToCheck: BaseEditor[] = [];
-			if (activeEditor) { editorsToCheck.push(activeEditor); }
-			visibleEditors.forEach(e => { if (e !== activeEditor) { editorsToCheck.push(e); } });
-			for (let i = 0; i < editorsToCheck.length; i++) {
-				const editorToCheck = editorsToCheck[i];
-				if (input.matches(editorToCheck.input)) {
-					return editorToCheck.position;
-				}
+			const group = this.stacks.findGroup(input, true);
+			if (null !== group) {
+				return this.stacks.positionOfGroup(group);
+			}
+		}
+
+		const config = this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>();
+		const revealIfOpen = config.workbench.editor.revealIfOpen;
+		// Respect option to reveal an editor if it is open (not necessarily visible)
+		const skipReuse = (options && options.index) || arg1;
+		if (!skipReuse && revealIfOpen) {
+			const group = this.stacks.findGroup(input, false);
+			if (null !== group) {
```

Comment:
OK, let me do the changes quickly as today I have a lot more bandwidth for this kind of work.